core/src/main/java/org/infinispan/conflict/impl/DefaultConflictManager.java

```diff
@@ -0,0 +1,416 @@
+      this.stateReceiver = stateReceiver;
+   }
+
+   @Start
+   public void start() {
+      this.cache = cache.withFlags(Flag.IGNORE_RETURN_VALUES, Flag.PUT_FOR_STATE_TRANSFER);
+      this.cache.addListener(this);
+      this.cacheName = cache.getName();
+      this.localAddress = rpcManager.getAddress();
+      this.installedTopology = distributionManager.getCacheTopology();
+
+      initRpcOptions();
+      cache.getCacheConfiguration().clustering()
+            .attributes().attribute(ClusteringConfiguration.REMOTE_TIMEOUT)
+            .addListener(((a, o) -> initRpcOptions()));
+   }
+
+   private void initRpcOptions() {
+      this.rpcOptions = rpcManager.getRpcOptionsBuilder(ResponseMode.SYNCHRONOUS_IGNORE_LEAVERS).build();
+   }
+
+   @Override
+   public void onTopologyUpdate(LocalizedCacheTopology cacheTopology) {
+      this.installedTopology = cacheTopology;
+      if (trace) log.tracef("Installed new topology %s: %s", cacheTopology.getTopologyId(), cacheTopology);
+   }
+
+   @Override
+   public void cancelVersionRequests() {
+      synchronized (versionRequestMap) {
+         versionRequestMap.values().forEach(VersionRequest::cancelRequestIfOutdated);
+      }
+   }
+
+   @Override
+   public void restartVersionRequests() {
+      VersionRequest request;
+      while ((request = retryQueue.poll()) != null) {
+         if (trace) log.tracef("Retrying %s", request);
+         request.start();
+      }
+   }
+
+   @Override
+   public Map<Address, InternalCacheValue<V>> getAllVersions(final K key) {
+      final VersionRequest request;
+      synchronized (versionRequestMap) {
+         request = versionRequestMap.computeIfAbsent(key, k -> new VersionRequest(k, stateConsumer.isStateTransferInProgress()));
+      }
+
+      try {
+         return request.completableFuture.get();
+      } catch (InterruptedException e) {
+         Thread.currentThread().interrupt();
+         throw new CacheException(e);
+      } catch (ExecutionException e) {
+         if (e.getCause() instanceof CacheException)
+            throw (CacheException) e.getCause();
+
+         throw new CacheException(e.getCause());
+      } finally {
+         synchronized (versionRequestMap) {
+            versionRequestMap.remove(key);
+         }
+      }
+   }
+
+   @Override
+   public Stream<Map<Address, CacheEntry<K, V>>> getConflicts() {
+      return getConflicts(installedTopology);
+   }
+
+   private Stream<Map<Address, CacheEntry<K, V>>> getConflicts(LocalizedCacheTopology topology) {
+      if (stateConsumer.isStateTransferInProgress())
+         throw log.getConflictsStateTransferInProgress(cacheName);
+
+      if (!streamInProgress.compareAndSet(false, true))
+         throw log.getConflictsAlreadyInProgress();
+
+      try {
+         return StreamSupport
+               .stream(new ReplicaSpliterator(topology), false)
+               .filter(filterConsistentEntries());
+      } finally {
+         streamInProgress.compareAndSet(true, false);
```

Comment:
You're returning the stream itself; no terminal operation was called so far on the stream, so it's still in progress, isn't it?